Small changes related to `VersionSpecificWorkerContextWrapper` and new tests

codeforgreen opened the pull request:
Related to #6505

What was done:
* updated all implementors of `IValidationSupport` to use `Logs.getTerminologyTroubleshootingLog()`
* added a bunch of new tests on `VersionSpecificWorkerContextWrapper` related to the cache and also testing the methods to fetch validation resources at different levels (unit and integration).

Note that the initial scope of this PR was changed and it now does not include any functional change.

michaelabuckley commented on hapi-fhir-docs/src/main/resources/ca/uhn/hapi/fhir/changelog/7_6_0/6505-validation-errors-profile-cache-expiry.yaml:
```diff
@@ -0,0 +1,8 @@
+would be returned which suggest the profile was not applied properly or used. 
+That happens because the validation cache entry expired within the $validate call 
+and the FHIR Core Validation library expects the same object instance for the same profile to be returned.
+This has been fixed by increasing the cache expiry to 10min."
```
question: Does this mean we are still vulnerable to rare failures if a cache entry expires in the middle of processing?

michaelabuckley commented on hapi-fhir-storage/src/main/java/ca/uhn/fhir/jpa/validation/ValidatorResourceFetcher.java:
```diff
@@ -105,10 +108,9 @@ private IBaseResource fetchByUrl(String url, IFhirResourceDao<?> dao, RequestDet
-				ourLog.warn(
-						String.format("Multiple results found for URL '%s', only the first will be considered.", url));
+		if (results != null && !results.isEmpty()) {
+			if (results.size() > 1 && ourLog.isWarnEnabled()) {
```
trivial: the ourLog.isWarnEnabled() check seems redundant since the call to warn uses a format string.

michaelabuckley commented on hapi-fhir-validation/src/main/java/org/hl7/fhir/common/hapi/validation/support/DefaultProfileValidationSupportNpmStrategy.java:
```diff
@@ -4,16 +4,16 @@
 
 public class DefaultProfileValidationSupportNpmStrategy extends NpmPackageValidationSupport {
-	private static final Logger ourLog = LoggerFactory.getLogger(DefaultProfileValidationSupportNpmStrategy.class);
+	private static final Logger ourLog = Logs.getTerminologyTroubleshootingLog();
```
Nice to see a new troubleshooting log.

codeforgreen commented:
> FYI @codeforgreen I believe this PR should be redundant once this merges (it also fixes the same issue by just never expiring StructureDefinions): #6508

I updated the scope of this PR to much smaller. I updated the description accordingly. It now just include the small compile warning or refactoring changes and additional tests. There is no change to the caches or validato, no changelog in this PR.
